### PR TITLE
OAK-9610: Have a jmx to explicitly expire indexing lease in mongo mk

### PR DESCRIPTION
**Pull request by tihom88**

### CODE REVIEW

File: oak-api/src/main/java/org/apache/jackrabbit/oak/api/jmx/package-info.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-@Version("4.9.0")
+@Version("4.10.0")
```
Comment (nit0906) [inline oak-api/src/main/java/org/apache/jackrabbit/oak/api/jmx/package-info.java:18]: Do we need to bump this, if the change is in a providerType interface ?

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java [hunk 2]
```diff
@@ -488,6 +488,14 @@ public void close() {
 
     private void runWhenPermitted() {
         if (indexStats.isPaused()) {
+            if (indexStats.forcedLeaseRelease){
+                try {
+                    clearLease();
+                } catch (CommitFailedException e) {
+                    log.warn("Unable to release lease, please try again");
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java]: Please log the exception (message and stack trace):

    log.warn("Unable to release lease, please try again", e);

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java [hunk 3]
```diff
@@ -626,6 +634,22 @@ private void runWhenPermitted() {
...
+    private void clearLease() throws CommitFailedException {
+        NodeState root = store.getRoot();
+        NodeState async = root.getChildNode(ASYNC);
+        String beforeCheckpoint = async.getString(name);
+        if (async.hasProperty(leasify(name))) {
+            NodeBuilder builder = root.builder();
+            builder.child(ASYNC).removeProperty(leasify(name));
+            mergeWithConcurrencyCheck(store, validatorProviders,
+                    builder, beforeCheckpoint, null, name);
+            log.debug("Lease property removed for lane: {}", name);
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java]: I would use log.info in this method (both cases), it is not called that often.

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java [hunk 4]
```diff
@@ -626,6 +634,22 @@ private void runWhenPermitted() {
         }
     }
 
+    private void clearLease() throws CommitFailedException {
+        NodeState root = store.getRoot();
+        NodeState async = root.getChildNode(ASYNC);
+        String beforeCheckpoint = async.getString(name);
+        if (async.hasProperty(leasify(name))) {
+            NodeBuilder builder = root.builder();
+            builder.child(ASYNC).removeProperty(leasify(name));
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java]: I would call leasify(name) only once.

**Review by thomasmueller — CHANGES_REQUESTED** [review]

File: oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdateLeaseRemoveTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                        // overridden close method so that lease is not deleted on async cycle completion.
+                        //super.close();
+                    }
+                };
+            }
+        };
+        async.setLeaseTimeOut(250);
+        async.run();
+        Long leaseValue = getLeaseValue(store);
+        assertTrue(leaseValue != null);
```
Comment (fabriziofortino) [inline oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdateLeaseRemoveTest.java]: this, and the same other 3 occurrences, can be simplified with
```suggestion
        assertNotNull(leaseValue);
```

**Review by fabriziofortino — APPROVED** [review]

File: oak-api/src/main/java/org/apache/jackrabbit/oak/api/jmx/package-info.java (hunk 1 above)
Comment (tihom88) [reply oak-api/src/main/java/org/apache/jackrabbit/oak/api/jmx/package-info.java:18]: Build is failing without minor bump. Adding providerType doesn't mean that version bump is not needed.

**Review by thomasmueller — APPROVED** [review]

**Pull request merged**